src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -53,9 +56,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+
+        const auto backgroundButton{ BackgroundButton() };
+        ToolTipService::SetToolTip(backgroundButton, box_value(RS_(L"ColorScheme_Background/Text")));
+        Automation::AutomationProperties::SetName(backgroundButton, RS_(L"ColorScheme_Background/Text"));
+
+        const auto cursorColorButton{ CursorColorButton() };
+        ToolTipService::SetToolTip(cursorColorButton, box_value(RS_(L"ColorScheme_CursorColor/Text")));
+        Automation::AutomationProperties::SetName(cursorColorButton, RS_(L"ColorScheme_CursorColor/Text"));
+
+        const auto selectionBackgroundButton{ SelectionBackgroundButton() };
+        ToolTipService::SetToolTip(selectionBackgroundButton, box_value(RS_(L"ColorScheme_SelectionBackground/Text")));
+        Automation::AutomationProperties::SetName(selectionBackgroundButton, RS_(L"ColorScheme_SelectionBackground/Text"));
+
+        const auto comboBox{ ColorSchemeComboBox() };
+        Automation::AutomationProperties::SetName(comboBox, RS_(L"ColorScheme_Name/Header"));
+        Automation::AutomationProperties::SetFullDescription(comboBox, RS_(L"ColorScheme_Name/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+        ToolTipService::SetToolTip(comboBox, box_value(RS_(L"ColorScheme_Name/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip")));
+
+        Automation::AutomationProperties::SetName(RenameButton(), RS_(L"Rename/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+
+        const auto nameBox{ NameBox() };
+        Automation::AutomationProperties::SetName(nameBox, RS_(L"ColorScheme_Name/Header"));
+        Automation::AutomationProperties::SetFullDescription(nameBox, RS_(L"ColorScheme_Name/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+        ToolTipService::SetToolTip(nameBox, box_value(RS_(L"ColorScheme_Name/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip")));
+
+        Automation::AutomationProperties::SetName(RenameAcceptButton(), RS_(L"RenameAccept/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+        Automation::AutomationProperties::SetName(RenameCancelButton(), RS_(L"RenameCancel/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+        Automation::AutomationProperties::SetName(AddNewButton(), RS_(L"ColorScheme_AddNewButton/Text"));
+        Automation::AutomationProperties::SetName(DeleteButton(), RS_(L"ColorScheme_DeleteButton/Text"));
+    }
+
+    void ColorSchemes::SizeChanged(IInspectable const& /*sender*/, Windows::UI::Xaml::SizeChangedEventArgs const& e)
+    {
+        // Record the width for ColorPanel when it's in horizontal mode
+        if (ColorPanel().Orientation() == Orientation::Horizontal)
+        {
+            const auto indentMargin{ Resources().Lookup(winrt::box_value(L"StandardIndentMargin")).as<Thickness>() };
+            const auto colorPanelWidth{ ColorPanel().ActualWidth() };
+            _colorPanelHorizontalWidth = colorPanelWidth + indentMargin.Left + indentMargin.Right;
+        }
```

Comment:
er this is going to record the horizontal width **every single time it changes**??